x/datapool/types/keys.go

```diff
@@ -54,11 +61,15 @@ func GetKeyPrefixDataValidateCert(poolID, round uint64, dataHash []byte) []byte
 	return append(GetKeyPrefixDataValidateCertByRound(poolID, round), dataHash...)
 }
 
-func GetKeyPrefixNFTRedeemReceiptByPoolID(poolID uint64, redeemer sdk.AccAddress) []byte {
-	keyPoolAppended := append(KeyPrefixNFTRedeemReceipts, sdk.Uint64ToBigEndian(poolID)...)
-	return append(keyPoolAppended, redeemer.Bytes()...)
+func GetKeyPrefixNFTRedeemReceiptByPoolID(poolID uint64) []byte {
+	return append(KeyPrefixNFTRedeemReceipts, sdk.Uint64ToBigEndian(poolID)...)
+}
+
+func GetKeyPrefixNFTRedeemReceipt(poolID, round, nftID uint64) []byte {
+	combineKeys := CombineKeys(GetKeyPrefixNFTRedeemReceiptByPoolID(poolID), sdk.Uint64ToBigEndian(round))
+	return CombineKeys(combineKeys, sdk.Uint64ToBigEndian(nftID))
```

Comment:
You can simply combine all the things :)
```suggestion
	return CombineKeys(GetKeyPrefixNFTRedeemReceiptByPoolID(poolID), sdk.Uint64ToBigEndian(round), sdk.Uint64ToBigEndian(nftID))
```